sd-cmd.go

```diff
@@ -28,46 +32,53 @@ func init() {
 	config.LoadConfig()
 }
 
+func runExecutor(sdAPI api.API, args []string) error {
+	err := executor.StartLog(args)
+	defer executor.FinishLog()
+
+	if err != nil {
+		return err
+	}
+	exec, err := executor.New(sdAPI, args)
+	if err != nil {
+		return err
+	}
+	err = exec.Run()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
 func runCommand(sdAPI api.API, args []string) error {
+	if len(os.Args) < minArgLength {
+		return fmt.Errorf("The argument num is not enough")
+	}
+
 	switch args[1] {
 	case "exec":
-		executor, err := executor.New(sdAPI, args[2:])
-		if err != nil {
-			return fmt.Errorf("Failed to create executor: %v", err)
-		}
-		output, err := executor.Run()
-		if err != nil {
-			fmt.Println(string(output))
-			return fmt.Errorf("Failed to run exec command: %v", err)
-		}
-		fmt.Println(string(output))
-		return nil
+		return runExecutor(sdAPI, args)
 	case "publish":
 		return fmt.Errorf("publish is not implemented yet")
 	case "promote":
 		return fmt.Errorf("promote is not implemented yet")
 	default:
-		return fmt.Errorf("No such type of command")
+		return runExecutor(sdAPI, args)
```

Comment:
Do we cause Error in case of `default`?

Reply:
https://github.com/screwdriver-cd/screwdriver/blob/master/design/commands.md#usage
We can execute without `exec` thus default call runExecutor function.

Reply:
I got it.